### PR TITLE
Runs pre-commit hooks during CI

### PR DESCRIPTION
**Pull request by bpkroth**
# Pull Request

## Title

Runs pre-commit hooks during CI
---

## Description

Ensures that the pre-commit hooks that would be executed locally are working as expected.

Keeping the other checks run as explicit steps for now as well.

---

## Type of Change

- 🧪 Tests

---

### CODE REVIEW

File: .github/workflows/pythonapp.yml [hunk 1]
```diff
@@ -60,6 +60,11 @@ jobs:
         else
           echo "All files have the required license headers."
         fi
+    - name: Check pre-commit hooks
+      run: |
+        python -m pip install pre-commit
```
Comment (ksaur) [inline .github/workflows/pythonapp.yml:65]: should we remove the lint check then?

Comment (bpkroth) [reply .github/workflows/pythonapp.yml:65]: We certainly could.

Comment (bpkroth) [reply .github/workflows/pythonapp.yml:65]: It'd be fewer places to maintain documentation, configs, etc. in sync.

Want me to make another PR?

Comment (ksaur) [reply .github/workflows/pythonapp.yml:65]: oh sorry i missed that it was merged! no worries we can leave it!

Comment (bpkroth) [reply .github/workflows/pythonapp.yml:65]: Actually noticed in #88 that it's running the checks in both now, so I might tweak or remove that anyways.  Future PR...

**Pull request merged**